Include log_toggler.py

OdatNurd opened the pull request:
This includes a simple plugin that allows you to easily toggle the state
of the various logging mechanisms in Sublime on and off using a key
binding, command palette entry or menu item.

This also serves as a simple demonstration of using the `is_checked` and
`description` methods of the command classes to customize how a command
appears in the menu.

As there is no API for determining the current state of the logging
flags, the command tracks the current state itself, which means that it
will get out of sync with reality if you use some external mechanism to
toggle the logging.

This will clear itself up the next time you invoke the command (although
it may appear to do nothing), but on the whole this is probably not that
important since you likely already know what you're doing in that case.

OdatNurd committed:
Include log_toggler.py

This includes a simple plugin that allows you to easily toggle the state
of the various logging mechanisms in Sublime on and off using a key
binding, command palette entry or menu item.

This also serves as a simple demonstration of using the `is_checked` and
`description` methods of the command classes to customize how a command
appears in the menu.

As there is no API for determining the current state of the logging
flags, the command tracks the current state itself, which means that it
will get out of sync with reality if you use some external mechanism to
toggle the logging.

This will clear itself up the next time you invoke the command (although
it may appear to do nothing), but on the whole this is probably not that
important since you likely already know what you're doing in that case.

keith-hall commented on plugins/README.md:
```diff
@@ -46,6 +46,13 @@ of each file for more information on full usage.
+ * [log_toggler.py](log_toggler.py) is an example `ApplicationCommand` that
+   makes it easier to toggle the state of logging for commands, input and
+   result regular expressions (useful for testing error capturing in a build
+   system) without having to drop the the console. It's also a demonstration on
```
small typo - you probably meant to say drop `to` the console

keith-hall commented on plugins/log_toggler.py:
```diff
@@ -0,0 +1,69 @@
+    """
+    Toggle the state of sublime's internal logging to the console based on the
+    log_type parameter, which should be one of the three sublime module API
+    endpoints, "log_command", "log_inoput" or "log_result_regex".
```
another typo in this docstring :)

keith-hall commented on plugins/log_toggler.py:
```diff
@@ -0,0 +1,69 @@
+_log_state = {
+    "log_commands": False,
+    "log_input": False,
+    "log_result_regex": False
```
I think it might be worth adding `log_build_systems` as well, and if `log_indexing` did anything useful, I would suggest that too

keith-hall reviewed (changes requested):
nice, I like how simple this plugin is, and how easy it is to add other logging toggle commands ;)
Let's fix a few small typos before we merge :)

OdatNurd committed:
Add a new logging type to the base plugin

This adds in an additional logging toggle for logging build system steps
and also includes some typo fixes and tweaks to some of the doc strings
and the README file.

OdatNurd commented:
I've added the oft forgotten and possibly undocumented logging of build systems and tweaked up the typos you mentioned. I also altered the documentation slightly to include the new option, so hopefully I didn't introduce anything new that my brain can't perceive. ;)
